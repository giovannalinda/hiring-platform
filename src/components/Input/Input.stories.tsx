import { Meta, Story } from '@storybook/react'
import { Input } from './Input'

export default {
  title: 'Input',
  component: Input,
} as Meta

export const Default: Story = () => <Input placeholder='example placeholder' />
