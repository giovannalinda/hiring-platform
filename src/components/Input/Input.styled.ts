import styled from 'styled-components'
import { theme } from '~/styles'

export const Container = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: row;
`

export const Button = styled.button`
  margin-left: -13.2rem;
  margin-top: 0.8rem;
  background: ${theme.colors.green200};
  height: 3.9rem;
  width: 12.8rem;
  border-radius: 30rem;
  font-weight: ${theme.font.weigths.bold};
  color: ${theme.colors.black};
`

export const GenericInput = styled.input`
  border: none;
  color: ${theme.colors.gray500};
  height: 5rem;
  padding: 0 6rem 0 ${theme.space[5]};
  margin-top: ${theme.space[1]};
  width: 100%;
  border-radius: 30rem;
  background: ${theme.colors.white};
  transition: all ${theme.transitions.slow};
  outline: 0;

  &:focus {
    outline-offset: 0.3rem;
    outline: 0.2rem solid ${theme.colors.green200};
  }

  ::placeholder {
    font-size: 1.2rem;
    color: ${theme.colors.gray200};
  }

  @media screen and (max-width: ${theme.breakpoints.tablet}) {
    width: 30rem;
  }

  @media screen and (max-width: ${theme.breakpoints.smallMobile}) {
    width: 25rem;
  }
`
