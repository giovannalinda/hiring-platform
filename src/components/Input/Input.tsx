import * as S from './Input.styled'

export function Input({ ...rest }) {
  return (
    <S.Container>
      <S.GenericInput {...rest} />
      <S.Button>Start Free Trial</S.Button>
    </S.Container>
  )
}
