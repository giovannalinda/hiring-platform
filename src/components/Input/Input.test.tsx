import { render } from '~/__helpers__/app-tests'
import { Input } from './Input'

describe('Input', () => {
  it('matches snapshot', () => {
    const { container } = render(<Input />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
