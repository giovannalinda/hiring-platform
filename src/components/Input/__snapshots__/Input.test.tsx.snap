// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<div
  class="sc-bczRLJ icfMzw"
>
  <input
    class="sc-dkzDqf dpHstQ"
  />
  <button
    class="sc-gsnTZi bjHqIv"
  >
    Start Free Trial
  </button>
</div>
`;
